godot-macros/src/godot_api.rs

```diff
@@ -308,11 +326,27 @@ where
             .expect("get_single_path_segment");
 
         let new_found = match attr_name {
-            name if name == "func" => Some(BoundAttr {
-                attr_name: attr_name.clone(),
-                index,
-                ty: BoundAttrType::Func(attr.value.clone()),
-            }),
+            name if name == "func" => {
+                // TODO you-win (August 8, 2023): handle default values here as well?
+
+                // Safe unwrap since #[func] must be present if we got to this point
```

Comment:
The code flow gets a bit weird (not your code, but the result of the existing one growing). 

We do this at the moment:
```rs
for attribute in all_attributes {
   match attribute {
       "func" => KvParser::parse(all_attributes) // again searching through all
       "signal" => ...
   }
}

detect_duplicates(); // custom validation that's less expressive than KvParser's built-in one
```

Whereas `KvParser` is designed to already extract the relevant attributes _and_ check for unhandled/duplicate/leftover ones in the `finish()` call. So you could do this:
```rs
if let Some(mut parser) = KvParser::parse(attributes, "func")? {
    // handle #[func]
    
    // eventually:
    parser.finish();
}

if let Some(_) = KvParser::parse(attributes, "signal")? {
    // handle #[signal]
}
```

Do you think this could be done?

Reply:
This is in progress, my latest WIP code does not work properly since it picks up other annotations first.

Currently, it's picking up `#[allow(clippy::uninlined_format_args)]`, but I will continue to poke at it.